aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -515,6 +515,368 @@ Tensor& matmul_out(Tensor &result, const Tensor & tensor1, const Tensor & tensor
+    x4 * I + x5 * A + x6 * A2 + x7 * A4);
+  return I + A + y2 * A2 + A8;
+}
+
+template <typename scalar_t>
+Tensor compute_T12(const Tensor& A) {
+  constexpr int num_prods = 4;
+  constexpr array2d<scalar_t, num_prods, num_prods> b = {{
+    {
+      9.0198e-16,
+      0.46932117595418237389,
+      -0.20099424927047284052,
+      -0.04623946134063071740
+    },
+    {
+      5.31597895759871264183,
+      1.19926790417132231573,
+      0.01179296240992997031,
+      0.01108844528519167989
+    },
+    {
+      0.18188869982170434744,
+      0.05502798439925399070,
+      0.09351590770535414968,
+      0.00610700528898058230
+    },
+    {
+      -2.0861320e-13,
+      -0.13181061013830184015,
+      -0.02027855540589259079,
+      -0.00675951846863086359
+    }
+  }};
+
+  const auto& I = at::eye(A.size(-1), A.options()).expand_as(A);
+  const auto& A2 = at::matmul(A, A);
+  const auto& A3 = at::matmul(A2, A);
+  std::array<
+    std::reference_wrapper<const Tensor>,
+    num_prods> As = {I, A, A2, A3};
+
+  std::array<Tensor, num_prods> Bs;
+  for (int i = 0; i < num_prods; ++i) {
+    Bs[i] = at::zeros(A.sizes(), A.options());
+  }
+
+  for (int i = 0; i < num_prods; ++i) {
+    for (int j = 0; j < num_prods; ++j) {
+      Bs[i] += b[i][j] * As[j];
+    }
+  }
+
+  const auto& A6 = Bs[2] + at::matmul(Bs[3], Bs[3]);
+  const auto& res = Bs[0] + at::matmul(Bs[1] + A6, A6);
+
+  return res;
+}
+
+template <typename scalar_t>
+Tensor compute_T18(const Tensor& A) {
+  constexpr int num_prods = 5;
+  constexpr array2d<scalar_t, num_prods, num_prods> b = {{
+    {
+      0.,
+      -1.00365581030144618291e-01,
+      -8.02924648241156932449e-03,
+      -8.92138498045729985177e-04,
+      0.
+    },
+    {
+      0.,
+      3.97849749499645077844e-01,
+      1.36783778460411720168e+00,
+      4.98289622525382669416e-01,
+      -6.37898194594723280150e-04
+    },
+    {
+      -1.09676396052962061844e+01,
+      1.68015813878906206114e+00,
+      5.71779846478865511061e-02,
+      -6.98210122488052056106e-03,
+      3.34975017086070470649e-05
+    },
+    {
+      -9.04316832390810593223e-02,
+      -6.76404519071381882256e-02,
+      6.75961301770459654925e-02,
+      2.95552570429315521194e-02,
+      -1.39180257516060693404e-05
+    },
+    {
+      0.,
+      0.,
+      -9.23364619367118555360e-02,
+      -1.69364939002081722752e-02,
+      -1.40086798182036094347e-05
+    }
+  }};
+
+  const auto& I = at::eye(A.size(-1), A.options()).expand_as(A);
+  const auto& A2 = at::matmul(A, A);
+  const auto& A3 = at::matmul(A2, A);
+  const auto& A6 = at::matmul(A3, A3);
+  std::array<
+    std::reference_wrapper<const Tensor>,
+    num_prods> As = {I, A, A2, A3, A6};
+
+  std::array<Tensor, num_prods> Bs;
+  for (int i = 0; i < num_prods; ++i) {
+    Bs[i] = at::zeros(A.sizes(), A.options());
+  }
+
+  for (int i = 0; i < num_prods; ++i) {
+    for (int j = 0; j < num_prods; ++j) {
+      Bs[i] += b[i][j] * As[j];
```

Comment:
Parts like these can be certainly fused for CUDA. Does it make sense to do the same for CPU?